Paket.LoadingScripts/LoadingScriptsGenerator.fs

```diff
@@ -0,0 +1,152 @@
+        depFile
+        |> Seq.map (fun p -> p.GroupName, handleParseState p.Packages)
+        |> Map.ofSeq
+        // generateScriptsFromDependenciesFile generateIncludeScript depFile
+
+    let testOrdering =
+      let testData =
+          [ { PackageResolver.ResolvedPackage.Name = PackageName("Test1")
+              PackageResolver.ResolvedPackage.Version = SemVer.Parse "1.0.0"
+              PackageResolver.ResolvedPackage.Dependencies =
+                Set.empty
+                |> Set.add(
+                    PackageName("other"), 
+                    VersionRequirement(VersionRange.Specific (SemVer.Parse "1.0.0"), PreReleaseStatus.No),
+                    Paket.Requirements.FrameworkRestrictions.Empty)
+              PackageResolver.ResolvedPackage.Unlisted = false
+              PackageResolver.ResolvedPackage.Settings = Requirements.InstallSettings.Default
+              PackageResolver.ResolvedPackage.Source = PackageSources.PackageSource.NuGetV2 { Url = ""; Authentication = None } }
+            { Name = PackageName("other")
+              Version = SemVer.Parse "1.0.0"
+              Dependencies = Set.empty
+              Unlisted = false
+              Settings = Requirements.InstallSettings.Default
+              Source = PackageSources.PackageSource.NuGetV2 { Url = ""; Authentication = None } }
+          ]
+      let result =
+        handleParseState testData
+        |> List.map (fun p -> p.Name)
+
+      System.Diagnostics.Debug.Assert(
+        result = 
+          [  PackageName("other")
+             PackageName("Test1")
+          ] : bool)
+          
+      let result2 =
+        handleParseState (testData |> List.rev)
+        |> List.map (fun p -> p.Name)
+
+      System.Diagnostics.Debug.Assert(
+        result2 = 
+          [  PackageName("other")
+             PackageName("Test1")
+          ] : bool)
+
+
+module ScriptGeneratingModule =
+  open System.IO
+  let getScriptName (package: PackageName) = sprintf "Include_%s.fsx" (package.GetCompareString())
+  let generateFSharpScript packagesOrGroupFolder (knownIncludeScripts:Map<PackageName, string>) (package: PackageResolver.ResolvedPackage) =
+    let packageFolder = Path.Combine (packagesOrGroupFolder, package.Name.GetCompareString())
+    let scriptFile = Path.Combine (packageFolder, getScriptName package.Name)
+    let relScriptFile = Path.Combine (package.Name.GetCompareString(), getScriptName package.Name)
+    let depLines =
+      package.Dependencies
+      |> Seq.map (fun (depName,_,_) -> sprintf "#load \"../%s\"" ((knownIncludeScripts |> Map.find depName).Replace("\\", "/")))
+
+    let dllFiles =
+      if package.Name.GetCompareString().ToLowerInvariant() = "fsharp.core" then
+        Seq.empty
+      else
+        let libDir = Path.Combine (packageFolder, "lib")
+        // TODO: Replace with some intelligent code to use the correct framework dlls,
+        // Generate multiple include scripts per-framework.
```

Comment:
I was more thinking of NET45, NET40, PROFILE111 and so forth, not a MONO one. I think writing a single script with `#if`s instead of multiple scripts per runtime is more difficult. But all the information we need should be available in paket.core somewhere :). 

I don't know if `#if`s are good from an end-user perspective as they would need to define their platform on the command line (because AFAIK there are no symbols predefined which we could use?). On the other side most users will probably understand what `Include_packagename_net45.fsx` or `Include_packagename_coreclr.fsx` means